### PR TITLE
Add Basic OTEL Tracing to TFBuddy

### PR DESCRIPTION
**Pull request by mplachter**
Add Tracing to TFBuddy

image

Currently waiting on this https://github.com/sl1pm4t/gongs/pull/3 to be merged than I'll update and sign DCO

### CODE REVIEW

**Comment by mplachter** [comment]
@sl1pm4t @davidwin93 if i fix this merge request is this good to merge?

File: cmd/root.go [hunk 1]
```diff
@@ -51,6 +56,25 @@ func init() {
...
+	otelPort := os.Getenv("TFBUDDY_OTEL_COLLECTOR_PORT")
+
+	opts := telemetry.Options{
+		Enabled:   enableOtel,
+		Host:      otelHost,
+		Port:      otelPort,
+		Version:   pkg.GitTag,
+		CommitSHA: pkg.GitCommit,
+	}
+	fmt.Printf("enabled: %v\thost: %s\tport: %s\n", enableOtel, otelHost, otelPort)
```
Comment (davidwin93) [inline cmd/root.go:71]: instead of `fmt` could these use `log`

**Comment by davidwin93** [comment]
If you resolve the merge conflicts I think this would be good to merge. Solid work!

File: pkg/gitlab_hooks/hook_worker.go [hunk 2]
```diff
@@ -39,14 +40,17 @@ func NewGitlabEventWorker(h *GitlabHooksHandler, js nats.JetStreamContext) *Gitl
 }
 
 func (w *GitlabEventWorker) processNoteEventStreamMsg(msg *NoteEventMsg) error {
+	ctx, span := otel.Tracer("hooks").Start(msg.Context, "CheckForMergeConflicts")
```
Comment (davidwin93) [inline pkg/gitlab_hooks/hook_worker.go:43]: Is CheckForMergeConflicts correct?

File: pkg/vcs/github/client.go [hunk 3]
```diff
@@ -265,57 +290,69 @@ func (c *Client) CloneMergeRequest(project string, mr vcs.MR, dest string) (vcs.
...
 
-func (c *Client) ResolveMergeRequestDiscussion(s string, i int, s2 string) error {
+func (c *Client) ResolveMergeRequestDiscussion(ctx context.Context, s string, i int, s2 string) error {
 	// This is a NoOp on GitHub
 	return nil
 }
 
-func (c *Client) AddMergeRequestDiscussionReply(prID int, fullName, discussionID, comment string) (vcs.MRNote, error) {
+func (c *Client) AddMergeRequestDiscussionReply(ctx context.Context, prID int, fullName, discussionID, comment string) (vcs.MRNote, error) {
+	ctx, span := otel.Tracer("TFC").Start(ctx, "ApplyMergeRequestDiscussionReply")
```
Comment (davidwin93) [inline pkg/vcs/github/client.go:305]: Was this meant to be AddMergeRequestDiscussionReply ?

**Review by davidwin93 — CHANGES_REQUESTED** [review]

File: pkg/vcs/github/client.go (hunk 3 above)
Comment (mplachter) [reply pkg/vcs/github/client.go:305]: Yeah we might have renamed the method good catch

**Review by davidwin93 — APPROVED** [review]

**Pull request merged**